Mathlib/MeasureTheory/Integral/PathIntegral/Basic.lean

```diff
@@ -0,0 +1,489 @@
+  letI : NormedSpace ℝ F := .restrictScalars ℝ 𝕜 F
+  simp only [pathIntegral, pathIntegralFun_add]
+  exact intervalIntegral.integral_add h₁ h₂
+
+theorem pathIntegral_fun_add (h₁ : PathIntegrable ω₁ γ) (h₂ : PathIntegrable ω₂ γ) :
+    ∫ᵖ x in γ, ω₁ x + ω₂ x = (∫ᵖ x in γ, ω₁ x) + ∫ᵖ x in γ, ω₂ x :=
+  pathIntegral_add h₁ h₂
+
+@[simp]
+theorem pathIntegralFun_zero : pathIntegralFun (0 : E → E →L[𝕜] F) γ = 0 := by
+  ext; simp [pathIntegralFun]
+
+@[simp]
+theorem pathIntegralFun_fun_zero : pathIntegralFun (fun _ ↦ 0 : E → E →L[𝕜] F) γ = 0 :=
+  pathIntegralFun_zero
+
+theorem PathIntegrable.zero : PathIntegrable (0 : E → E →L[𝕜] F) γ := by
+  simp [PathIntegrable, IntervalIntegrable.zero]
+
+theorem PathIntegrable.fun_zero : PathIntegrable (fun _ ↦ 0 : E → E →L[𝕜] F) γ := .zero
+
+@[simp]
+theorem pathIntegral_zero : pathIntegral (0 : E → E →L[𝕜] F) γ = 0 := by simp [pathIntegral]
+
+@[simp]
+theorem pathIntegral_fun_zero : ∫ᵖ _ in γ, (0 : E →L[𝕜] F) = 0 := pathIntegral_zero
+
+@[simp]
+theorem pathIntegralFun_neg : pathIntegralFun (-ω) γ = -pathIntegralFun ω γ := by
+  ext; simp [pathIntegralFun]
+
+nonrec theorem PathIntegrable.neg (h : PathIntegrable ω γ) : PathIntegrable (-ω) γ := by
+  simpa [PathIntegrable] using h.neg
+
+theorem PathIntegrable.fun_neg (h : PathIntegrable ω γ) : PathIntegrable (-ω ·) γ :=
+  h.neg
+
+@[simp]
+theorem PathIntegrable.neg_iff : PathIntegrable (-ω) γ ↔ PathIntegrable ω γ :=
+  ⟨fun h ↦ by simpa using h.neg, .neg⟩
+
+@[simp]
+theorem PathIntegrable.fun_neg_iff : PathIntegrable (-ω ·) γ ↔ PathIntegrable ω γ :=
+  PathIntegrable.neg_iff
+
+@[simp]
+theorem pathIntegral_neg : pathIntegral (-ω) γ = -∫ᵖ x in γ, ω x := by
+  simp [pathIntegral]
+
+@[simp]
+theorem pathIntegral_fun_neg : ∫ᵖ x in γ, -ω x = -∫ᵖ x in γ, ω x := pathIntegral_neg
+
+@[simp]
+theorem pathIntegralFun_sub :
+    pathIntegralFun (ω₁ - ω₂) γ = pathIntegralFun ω₁ γ - pathIntegralFun ω₂ γ := by
+  simp [sub_eq_add_neg]
+
+protected nonrec theorem PathIntegrable.sub (h₁ : PathIntegrable ω₁ γ) (h₂ : PathIntegrable ω₂ γ) :
+    PathIntegrable (ω₁ - ω₂) γ :=
+  sub_eq_add_neg ω₁ ω₂ ▸ h₁.add h₂.neg
+
+theorem pathIntegral_sub (h₁ : PathIntegrable ω₁ γ) (h₂ : PathIntegrable ω₂ γ) :
+    pathIntegral (ω₁ - ω₂) γ = (∫ᵖ x in γ, ω₁ x) - ∫ᵖ x in γ, ω₂ x := by
+  rw [sub_eq_add_neg, sub_eq_add_neg, pathIntegral_add h₁ h₂.neg, pathIntegral_neg]
+
+theorem pathIntegral_fun_sub (h₁ : PathIntegrable ω₁ γ) (h₂ : PathIntegrable ω₂ γ) :
+    ∫ᵖ x in γ, ω₁ x - ω₂ x = (∫ᵖ x in γ, ω₁ x) - ∫ᵖ x in γ, ω₂ x :=
+  pathIntegral_sub h₁ h₂
+
+
+section RestrictScalars
+
+variable {𝕝 : Type*} [RCLike 𝕝] [NormedSpace 𝕝 F] [NormedSpace 𝕝 E]
+  [LinearMap.CompatibleSMul E F 𝕝 𝕜]
+
+@[simp]
+theorem pathIntegralFun_restrictScalars :
+    pathIntegralFun (fun t ↦ (ω t).restrictScalars 𝕝) γ = pathIntegralFun ω γ := by
+  ext
+  letI : NormedSpace ℝ E := .restrictScalars ℝ 𝕜 E
+  simp [pathIntegralFun_def]
+
+@[simp]
+theorem PathIntegrable.restrictScalars_iff :
```

Comment:
```suggestion
theorem pathIntegrable_restrictScalars_iff :
```
Also, register the direction from right to left with an alias.

Reply:
I saw `MyNamespace.myProp_iff` here and there and someone told me that this is a Lean 4 style (can't find it now). I'm OK with either naming scheme.